---
title: "Data analysis"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup

```{r message=FALSE}
library(tidyverse)
library(vegan)
library(pheatbuilder)
library(ggbeeswarm)
library(broom)
library(lmerTest)
library(broom.mixed)
library(usedist)
library(adonisplus)
library(mirix)
```

## Functions

```{r}
taxonomic_ranks <- c(
  "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

split_assignments <- function (assignments, ranks=taxonomic_ranks, split="; ",
                               remove = "^[kpcofgs]__") {
  n <- length(ranks)
  res <- stringr::str_split_fixed(assignments, pattern = split, n)
  colnames(res) <- ranks
  res <- tibble::as_tibble(res)

  if (!is.null(remove)) {
    res <- dplyr::mutate(res, dplyr::across(ranks, stringr::str_remove, remove))
  }

  dplyr::mutate(res, dplyr::across(ranks, dplyr::na_if, ""))
}

simplify_assignments <- function (assignments_df, rank1="Phylum",
                                  rank2="Genus") {
  if (is.character(rank1)) {
    rank1 <- match(rank1, colnames(assignments_df))
  }
  if (is.character(rank2)) {
    rank2 <- match(rank2, colnames(assignments_df))
  }
  apply(assignments_df, 1, function (x) {
    x <- na.omit(as.character(x))
    n <- length(x)
    if (n == 1)     return(x)
    if (n < rank1)  return(paste(x, collapse=" "))
    if (n == rank1) return(x[rank1])
    if (n < rank2)  return(paste(x[rank1], "-", x[length(x)]))
    return(paste(x[rank1], "-", x[rank2]))
  })
}
```

```{r}
filter_taxa_shotgun <- function (taxa_df) {
  taxa_df %>%
    filter(Kingdom %in% "Bacteria")
}

filter_taxa_16S <- function (taxa_df) {
  taxa_df %>%
    filter(!is.na(Phylum)) %>%
    filter(!(Class %in% "Chloroplast")) %>%
    filter(!(Family %in% "mitochondria"))
}
```


```{r}
save_figure <- function (fbase, width, height, p = last_plot()) {
  ggsave(
    paste0(fbase, ".pdf"), plot = p, width = width, height = height,
    useDingbats = FALSE)
  ggsave(
    paste0(fbase, ".png"), plot = p, width = width, height = height,
    dpi = 300)
}
```



## Colors

```{r}
resistant_colors <- c(resistant = "#ff8585", susceptible = "#999999")
```

```{r}
group_colors_placebo <- c(Placebo = "#1F77B4", Vancomycin = "#FF7F0E")
group_colors_control <- c(Control = "#1F77B4", Amoxicillin = "#FF7F0E")
week_colors <- c(
  "Week 0" = "#A6BDDB", "Week 4" = "#74A9CF", "Week 8" = "#0570B0",
  "Week 12" = "#045A8D")
day_colors <- c(
  "Day 0" = "#A6BDDB", "Day 1" = "#74A9CF", "Day 3" = "#0570B0",
  "Day 7" = "#045A8D")
day_colors_long <- c(
  "Day 0" = "#A6BDDB", "Day 7" = "#74A9CF", "Day 14" = "#3690C0",
  "Day 21" = "#0570B0", "Day 28" = "#045A8D")
```

```{r}
taxa_colors <- c(
  "Actinobacteria - Bifidobacterium" = "#d7b3f5",
  "Bacteroidetes - Bacteroides" = "#045A8D",
  "Bacteroidetes - Bacteroides caccae" = "#A6BDDB",
  "Bacteroidetes - Bacteroides dorei" = "#74A9CF",
  "Bacteroidetes - Bacteroides ovatus" = "#0570B0",
  "Bacteroidetes - Bacteroides thetaiotaomicron" = "#1ac3ed",
  "Bacteroidetes - Bacteroides vulgatus" = "#b5e0eb",
  "Firmicutes - Clostridiales" = "#f5e173",
  "Firmicutes - Lachnospiraceae" = "#f2bb24",
  "Firmicutes - Ruminococcaceae" = "#c5fa8c",
  "Firmicutes - Anaerostipes [Eubacterium] rectale" = "#a68428",
  "Firmicutes - Blautia" = "#ffc421",
  "Firmicutes - Enterococcus" = "#8e1496",
  "Firmicutes - Faecalibacterium prausnitzii" = "#d1cc7d",
  "Firmicutes - Flavonifractor plautii" = "#a0c497",
  "Firmicutes - Lachnoclostridium sp. YL32" = "#92f098",
  "Firmicutes - Lactobacillus" = "#f5a573",
  "Firmicutes - Streptococcus" = "#f59396",
  "Proteobacteria - Enterobacteriaceae" = "#bf0a8f",
  "Proteobacteria - Pseudomonas tolaasii" = "#f593db",
  "Verrucomicrobia - Akkermansia muciniphila" = "#1cad34",
  "Other" = "#BBBBBB")
```

# Basolo (vancomycin, shotgun)

## Import

```{r}
s_basolo <- read_tsv("basolo_samples.tsv")
taxa_basolo <- read_tsv("basolo_taxa.tsv")
cts_basolo <- read_tsv("basolo_cts.tsv")
```

## Alpha diversity

```{r}
adiv_basolo <- cts_basolo %>%
  group_by(sample_id) %>%
  summarize(
    total_reads = sum(reads),
    richness = vegan::rarefy(reads, 1000),
    shannon = vegan::diversity(reads))
```

```{r}
adiv_basolo %>%
  left_join(s_basolo, by = "sample_id") %>%
  ggplot(aes(x = study_group, y = shannon, color = study_group)) +
  geom_quasirandom() +
  scale_color_manual(values = group_colors_placebo, guide = "none") +
  labs(y = "Shannon diversity", x = "") +
  theme_bw()
```

```{r}
adiv_basolo %>%
  left_join(s_basolo, by = "sample_id") %>%
  summarise(tidy(lm(shannon ~ study_group, data = .)))
```

## Taxonomic abundance

```{r}
diffabund_basolo <- cts_basolo %>%
  group_by(sample_id) %>%
  mutate(prop = reads / sum(reads)) %>%
  ungroup() %>%
  group_by(taxon_id) %>%
  filter(mean(prop) > 0.005) %>%
  ungroup() %>%
  left_join(taxa_basolo, by = "taxon_id") %>%
  left_join(s_basolo, by = "sample_id") %>%
  mutate(log_prop = log10(prop + 1e-6)) %>%
  nest_by(assignment) %>%
  summarise(tidy(lm(log_prop ~ study_group, data = data))) %>%
  ungroup() %>%
  filter(term %in% "study_groupVancomycin") %>%
  mutate(fdr = p.adjust(p.value, method = "fdr")) %>%
  arrange(p.value) %>%
  select(assignment, estimate, fdr)
```

```{r}
cts_basolo %>%
  group_by(sample_id) %>%
  mutate(prop = reads / sum(reads)) %>%
  ungroup() %>%
  group_by(taxon_id) %>%
  filter(mean(prop) > 0.005) %>%
  ungroup() %>%
  left_join(taxa_basolo, by = "taxon_id") %>%
  left_join(s_basolo, by = "sample_id") %>%
  ggplot(aes(x = study_group, y = prop, color = study_group)) +
  geom_quasirandom() +
  facet_wrap(~ assignment) +
  scale_y_log10() +
  scale_color_manual(values = group_colors_placebo, guide = "none") +
  labs(y = "Relative abundance", x = "") +
  theme_bw()
```

# Willmann (ciprofloxacin, shotgun)

## Import

```{r}
s_willmann <- read_tsv("willmann_samples.tsv")
taxa_willmann <- read_tsv("willmann_taxa.tsv")
cts_willmann <- read_tsv("willmann_cts.tsv")
```

```{r}
card_info <- read_tsv("card_info.tsv")
card_header_info <- read_tsv("card_header_info.tsv")
```

## Alpha Diversity

```{r warning=FALSE}
adiv_willmann <- cts_willmann %>%
  group_by(sample_id) %>%
  summarize(
    total_reads = sum(reads),
    richness = vegan::rarefy(reads, 1000),
    shannon = vegan::diversity(prop))
```

```{r}
s_willmann %>%
  left_join(adiv_willmann, by = "sample_id") %>%
  ggplot(aes(x = study_day, y = shannon, color = study_day)) +
  geom_quasirandom() +
  scale_color_manual(values = day_colors, guide = "none") +
  labs(y = "Shannon diversity", x = "") +
  theme_bw()
```

```{r}
s_willmann %>%
  left_join(adiv_willmann, by = "sample_id") %>%
  lmer(shannon ~ study_day + (1|subject_id), data = .) %>%
  summary()
```

## Taxonomic abundance

```{r}
diffabund_willmann <- cts_willmann %>%
  left_join(taxa_willmann, by = "taxon_id") %>%
  group_by(sample_id, assignment) %>%
  summarise(prop = sum(prop), .groups = "drop") %>%
  group_by(assignment) %>%
  filter(mean(prop) > 0.005) %>%
  ungroup() %>%
  left_join(s_willmann, by = "sample_id") %>%
  mutate(log_prop = log10(prop + 1e-6)) %>%
  nest_by(assignment) %>%
  summarise(
    tidy(lmer(log_prop ~ study_day + (1|subject_id), data = data)),
    .groups = "drop") %>%
  filter(effect %in% "fixed", str_starts(term, "study_day")) %>%
  mutate(term = str_remove(term, "study_day")) %>%
  group_by(term) %>%
  mutate(fdr = p.adjust(p.value, method = "fdr")) %>%
  ungroup() %>%
  arrange(term, p.value) %>%
  filter(fdr < 0.05) %>%
  select(assignment, term, estimate, fdr)
write_csv(diffabund_willmann, "tables/supplementary_table_3.csv")
```



## CARD genes

```{r}
card_props_willmann <- adiv_willmann %>%
  left_join(card_willmann, by = "sample_id") %>%
  mutate(prop = reads / total_reads) %>%
  left_join(card_info, by = "accession") %>%
  filter(str_detect(drug_class, "fluoroquinolone")) %>%
  left_join(card_header_info, by = "accession") %>%
  filter(!is.na(aro)) %>%
  left_join(s_willmann, by = "sample_id") %>%
  group_by(accession) %>%
  filter(mean(prop) > 1e-5) %>%
  ungroup() %>%
  mutate(species = str_remove(species, "(?<=K-12).+")) %>%
  mutate(species = str_remove(species, "(?<=enterica) subsp.+")) %>%
  mutate(species = str_remove(species, "(?<=doylei).+")) %>%
  mutate(species = str_remove(species, "subsp. aureus ")) %>%
  mutate(species = str_remove(species, "(?<=damselae).+"))
```

```{r}
card_props_willmann %>%
  group_by(accession) %>%
  summarise(mean_prop = mean(prop), .groups = "drop") %>%
  arrange(desc(mean_prop))
```

```{r}
card_tests_willman <- card_props_willmann %>%
  mutate(log_prop = log10(prop + 1e-6)) %>%
  group_by(aro, gene_name, resistance_mechanism, species) %>%
  summarize(
    tidy(lmer(log_prop ~ study_day + (1|subject_id))),
    .groups = "drop") %>%
  filter(str_detect(term, "study_day")) %>%
  mutate(term = str_remove(term, "study_day")) %>%
  group_by(term) %>%
  mutate(fdr = p.adjust(p.value)) %>%
  ungroup() %>%
  rename(study_day = term) %>%
  mutate(sig_label = p_stars(fdr))
card_tests_willman %>%
  filter(fdr < 0.05) %>%
  select(aro, study_day, estimate, fdr)
```

```{r}
card_props_willmann %>%
  ggplot() +
  ggbeeswarm::geom_quasirandom(
    aes(x = study_day, color = study_day, y = prop)) +
  geom_text(
    aes(label = sig_label, x = study_day, y = 0.001), 
    data = dplyr::filter(card_tests_willman, fdr < 0.05)) +
  facet_wrap(~ resistance_mechanism + aro + gene_name + species) +
  scale_y_log10() +
  scale_color_manual(values = day_colors) +
  labs(y = "Relative abundance", x = "") +
  theme_bw() +
  theme(
    strip.background = element_blank(),
    legend.position = "none")
save_figure("figures/mirix_figS1", width = 20, height = 20)
```

